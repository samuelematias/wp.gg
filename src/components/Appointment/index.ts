export {
    Appointment,
    AppointementProps,
} from './Appointment';
