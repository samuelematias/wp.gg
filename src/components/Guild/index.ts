export {
    Guild,
    GuildProps,
} from './Guild';
