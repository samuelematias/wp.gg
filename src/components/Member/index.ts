export {
    Member,
    MemberProps,
} from './Member';
