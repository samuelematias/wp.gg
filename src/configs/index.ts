export {
    COLLECTION_USERS,
    COLLECTION_APPOINTMENTS
} from './database';
