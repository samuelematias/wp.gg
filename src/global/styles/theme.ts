export const theme = {
        colors: {
                primary: '#9146FF',

                secondary100: '#0E0E0E',
                secondary90: '#0D133D',
                secondary80: '#0E1647',
                secondary85: '#171F52',
                secondary70: '#1B2565',
                secondary60: '#1B2565',
                secondary50: '#243189',
                secondary40: '#1D2766',
                secondary30: '#495BCC',

                overlay: 'rgba(0,0,0,0.7)',
                highlight: '#ABB1CC',
                heading: '#DDE3F0',
                line: '#0E0E0E',
                on: '#32BD50',
        },

        fonts: {
                title700: 'Inter_700Bold',
                title600: 'Inter_600SemiBold',
                text400: 'Inter_400Regular',
                text500: 'Inter_500Medium',
        }
};
