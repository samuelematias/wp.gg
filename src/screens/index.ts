export * from './AppointmentCreate';
export * from './AppointmentDetails';
export * from './Guilds';
export * from './Home';
export * from './SignIn';
