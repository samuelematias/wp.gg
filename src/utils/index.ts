export { categories } from './categories';
export { IMAGE_PLACEHOLDER } from './constants';
