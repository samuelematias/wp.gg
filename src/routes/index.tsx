import {
    DefaultTheme,
    NavigationContainer
} from '@react-navigation/native';

import {
    SignIn
} from '../screens';

import { useAuth } from '../hooks';

import { AppRoutes } from './app.routes';

import { theme } from '../global/styles';

export function Routes() {
    const { user } = useAuth();

    const navTheme = {
        ...DefaultTheme,
        colors: {
            ...DefaultTheme.colors,
            background: theme.colors.secondary100,
        },
    };
    return (
        <NavigationContainer
            theme={navTheme}
        >
            {user?.id ? <AppRoutes /> : <SignIn />}
        </NavigationContainer>
    )
}
