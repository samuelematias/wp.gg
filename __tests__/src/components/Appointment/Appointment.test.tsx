import 'react-native';
import React from 'react';

import {
    render,
    fireEvent,
} from '../../../../jest/helper';

import { Appointment } from '../../../../src/components';

import { theme } from '../../../../src/global/styles';

const onPressMock = jest.fn();

const appointment = {
    id: '1',
    guild: {
        id: '1',
        name: 'Lendários',
        icon: null,
        owner: true,
    },
    category: '1',
    date: '22/06 às 20:40h',
    description: 'É hoje que vamos chegar ao challenger sem perder uma partida da md10',
};

const appointmentTestId = {
    touchable: 'appointment',
    playerIcon: 'appointment-player-icon',
};

const guildIconTestId = {
    guildImage: 'guild-icon-image',
    guildImagePlaceholder: 'guild-icon-image-placeholder',
};

const renderAppointment =
    <Appointment
        data={appointment}
        onPress={onPressMock}
    />;

describe('Appointment', () => {

    it('should render correctly and match the snapshot', () => {
        const { toJSON } = render(renderAppointment);

        expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly', () => {
        const { getByText } = render(renderAppointment);

        expect(getByText(appointment.guild.name)).toBeTruthy();
    });

    it('should call onPress when pressed', () => {
        const { getByTestId } = render(renderAppointment);

        fireEvent.press(getByTestId(appointmentTestId.touchable));

        expect(onPressMock).toHaveBeenCalled();
    });

    it('should render correctly the Anfitrião, when owner', () => {
        const { getByText } = render(renderAppointment);

        expect(getByText('Anfitrião')).toBeTruthy();
    });

    it('should render correctly the playerTitle color, theme.colors.primary, when owner', () => {
        const { getByText } = render(renderAppointment);

        const playerTitle = getByText('Anfitrião');

        expect(playerTitle.props.style[0].color).toBe(theme.colors.primary);

    });

    it('should render correctly the Visitante, when not owner', () => {
        const appointmentNotOwner = {
            ...appointment,
            guild: {
                ...appointment.guild,
                owner: false,
            },
        };

        const renderAppointmentNotOwner =
            <Appointment
                data={appointmentNotOwner}
                onPress={onPressMock}
            />;

        const { queryByText } = render(renderAppointmentNotOwner);

        expect(queryByText('Visitante')).toBeTruthy();
    });

    it('should render correctly the playerTitle color, theme.colors.on, when not owner', () => {
        const appointmentNotOwner = {
            ...appointment,
            guild: {
                ...appointment.guild,
                owner: false,
            },
        };

        const renderAppointmentNotOwner =
            <Appointment
                data={appointmentNotOwner}
                onPress={onPressMock}
            />;

        const { getByText } = render(renderAppointmentNotOwner);

        const playerTitle = getByText('Visitante');

        expect(playerTitle.props.style[0].color).toBe(theme.colors.on);
    });

    it('should render correctly the icon color, theme.colors.on, when not owner', () => {
        const appointmentHasIcon = {
            ...appointment,
            guild: {
                ...appointment.guild,
                owner: false,
            },
        };

        const renderAppointmentNotOwner =
            <Appointment
                data={appointmentHasIcon}
                onPress={onPressMock}
            />;

        const { getByTestId } = render(renderAppointmentNotOwner);

        const playerIcon = getByTestId(appointmentTestId.playerIcon);

        expect(playerIcon.props.fill).toBe(theme.colors.on);
    });

    it('should render correctly the icon color, theme.colors.primary, when owner', () => {
        const { getByTestId } = render(renderAppointment);

        const playerIcon = getByTestId(appointmentTestId.playerIcon);

        expect(playerIcon.props.fill).toBe(theme.colors.primary);
    });

    it('should render correctly when has GuildIcon image', () => {
        const appointmentHasIcon = {
            ...appointment,
            guild: {
                ...appointment.guild,
                icon: 'image.png',
                owner: false,
            },
        };

        const renderAppointmentHasGuildIcon =
            <Appointment
                data={appointmentHasIcon}
                onPress={onPressMock}
            />;

        const { getByTestId } = render(renderAppointmentHasGuildIcon);

        expect(getByTestId(guildIconTestId.guildImage)).toBeTruthy();
    });

    it('should render correctly when has no GuildIcon image', () => {
        const { queryByTestId } = render(renderAppointment);

        expect(queryByTestId(guildIconTestId.guildImage)).toBeFalsy();
    });

    it('should render correctly the placeholder, when has no image', () => {
        const { queryByTestId } = render(renderAppointment);

        expect(queryByTestId(guildIconTestId.guildImagePlaceholder)).toBeTruthy();
    });

    it('should render correctly the category', () => {
        const { getByText } = render(renderAppointment);

        expect(getByText('Ranqueada')).toBeTruthy();
    });

    // w8 for this feature implementation.
    /* it('should be able to render with selected', () => {
*     const { getByText, getByTestId } = render(<Appointment data={appointment} />);

*     fireEvent(getByTestId(appointmentTestId.touchable), 'onLongPress');

*     expect(getByText('Desmarcar')).toBeTruthy();
* }); */
})
