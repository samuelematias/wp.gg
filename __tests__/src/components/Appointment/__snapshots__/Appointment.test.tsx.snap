// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment should render correctly and match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "flexDirection": "row",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="Appointment.testId"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9146FF",
          "borderRadius": 8,
          "height": 62,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 62,
        },
      ]
    }
  >
    <svg
      data-file-name="SvgDiscord"
      style={
        Array [
          Object {
            "height": 40,
            "width": 40,
          },
        ]
      }
      testID="GuildIcon.image.placeholder.testId"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingLeft": 20,
          "paddingTop": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#DDE3F0",
              "fontFamily": "Inter_700Bold",
              "fontSize": 18,
            },
          ]
        }
      >
        Lendários
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#ABB1CC",
              "fontFamily": "Inter_400Regular",
              "fontSize": 13,
              "marginRight": 24,
            },
          ]
        }
      >
        Ranqueada
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <svg
          data-file-name="SvgCalendar"
          fill="#9146FF"
        />
        <Text
          style={
            Array [
              Object {
                "color": "#ABB1CC",
                "fontFamily": "Inter_400Regular",
                "fontSize": 13,
                "marginLeft": 7,
              },
            ]
          }
        >
          22/06 às 20:40h
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
          ]
        }
      >
        <svg
          data-file-name="SvgPlayer"
          fill="#9146FF"
          testID="Appointment.player.icon.testId"
        />
        <Text
          owner={true}
          style={
            Array [
              Object {
                "color": "#9146FF",
                "fontFamily": "Inter_500Medium",
                "fontSize": 13,
                "marginLeft": 7,
                "marginRight": 24,
              },
            ]
          }
        >
          Anfitrião
        </Text>
      </View>
    </View>
  </View>
</View>
`;
